api/src/main/java/io/knotx/server/api/handler/RequestEventHandlerResult.java

```diff
@@ -90,7 +91,7 @@ public Buffer getBody() {
   @Override
   public String toString() {
     return "RequestEventHandlerResult{" +
-        "requestEvent=" + requestEvent.toJson() +
+        "requestEvent=" + getRequestEvent().map(RequestEvent::toJson).orElseGet(JsonObject::new) +
```

Comment:
Please add toString() method in RequstEvent and then call toJson() inside. Then in RequestEventHandlerResult we would see null as a value.